File: spring/actuator/src/main/java/com/hyfly/actuator/entity/message/ChangeSummary.java
package com.hyfly.actuator.entity.message;

import com.alibaba.fastjson2.annotation.JSONField;
import lombok.Data;

@Data
public class ChangeSummary {

    public static final String TYPE_SUFFIX = "_summary";

    public static final String CHANGE_TYPE = "change" + TYPE_SUFFIX;

    @JSONField(name = "add")
    private Integer addCount;

    @JSONField(name = "change")
    private Integer changeCount;

    @JSONField(name = "remove")
    private Integer removeCount;

    @JSONField(name = "import")
    private Integer importCount;

    /**
     * apply, plan, destroy
     * {@link com.hyfly.actuator.entity.message.constants.SummaryOperation}
     */
    @JSONField(name = "operation")
    private String operation;
}

File: terraform/src/main/kotlin/com/hyfly/tf/actuator/entity/message/ChangeSummary.kt
package com.hyfly.tf.actuator.entity.message

import com.alibaba.fastjson2.annotation.JSONField

class ChangeSummary {
    @JSONField(name = "add")
    var addCount: Int? = null

    @JSONField(name = "change")
    var changeCount: Int? = null

    @JSONField(name = "remove")
    var removeCount: Int? = null

    @JSONField(name = "import")
    var importCount: Int? = null

    /**
     * apply, plan, destroy
     */
    @JSONField(name = "operation")
    var operation: String? = null

    companion object {
        const val TYPE_SUFFIX: String = "_summary"

        const val CHANGE_TYPE: String = "change" + TYPE_SUFFIX
    }
}
